lib/src/st2110/st_rx_video_session.c

```diff
@@ -1526,9 +1526,26 @@ static int rv_handle_frame_pkt(struct st_rx_video_session_impl* s, struct rte_mb
   bool exist_ts = false;
   struct st_rx_video_slot_impl* slot = rv_slot_by_tmstamp(s, tmstamp, NULL, &exist_ts);
   if (!slot || !slot->frame) {
+    /* exists_ts is true when the target slot is found by timestamp */
     if (exist_ts) {
-      s->stat_pkts_redundant_dropped++;
-      slot->pkts_recv_per_port[s_port]++;
+      /* Check if the packet has been already received and the bitmap is allocated by
+       * timestamp */
+      if (slot->seq_id_got && slot->frame_bitmap) {
```

Comment:
I don't like the fact that here we are dereferencing a slot right after checking that it can be == NULL. Please rewrite these logic in a way that will prevent secfaults.